pm/sendermonitor.go

```diff
@@ -291,17 +289,20 @@ func (sm *LocalSenderMonitor) startTicketQueueConsumerLoop(queue *ticketQueue, d
 		select {
 		case red := <-queue.Redeemable():
 			tx, err := sm.redeemWinningTicket(red.SignedTicket)
-			if err != nil {
-				red.resCh <- struct {
-					txHash ethcommon.Hash
-					err    error
-				}{ethcommon.Hash{}, err}
-			} else {
-				red.resCh <- struct {
-					txHash ethcommon.Hash
-					err    error
-				}{tx.Hash(), nil}
+			res := struct {
+				txHash ethcommon.Hash
+				err    error
+			}{
+				ethcommon.Hash{},
+				err,
+			}
+			// FIXME: If there are replacement txs then tx.Hash() could be different
```

Comment:
Note: Create an issue to track this.

This was a problem before this PR and more substantial changes will be required to address it (i.e. return the tx that was mined from `CheckTx()`). The negative impact is relatively minimal though - the DB will just end up tracking the wrong tx hash for a redeemed ticket if a replacement tx was mined.